### ISSUE TITLE
Make ID generation per-worker make it lock-free

### ISSUE THREAD

**Issue by arpitbbhayani** [DiceDB/dice#869]

**Comment by ayushsatyam146** [comment]
I would like to take this up @arpitbbhayani @JyotinderSingh @lucifercr07. Might need to learn a few things but it would help me gain more insights about the project

**Comment by JyotinderSingh** [comment]
> I would like to take this up @arpitbbhayani @JyotinderSingh @lucifercr07. Might need to learn a few things but it would help me gain more insights about the project

I believe you already have a few open issues assigned to you. Till then I would like to keep this open for others to pick up. Feel free to take this up once you're done with other tasks (and if this issue is available)

**Comment by gauravrp** [comment]
@JyotinderSingh any chance i can pick this up, in collaboration with someone else?
About to wrap up with another issue that's assigned to me

**Comment by mohdammar128** [comment]
Can you assign it to me @arpitbbhayani .I would like to work on this .

**Comment by arpitbbhayani** [comment]
@ayushsatyam146 go for it. assigned.

Few pointers

- make sure there are no locks
- the ID needs to be unique across worker threads
- you might need to incorporate worker ID in the generated ID

**Comment by gauravrp** [comment]
@ayushsatyam146 can we work together on this?
Would love to understand a thing or two on this topic

**Comment by ayushsatyam146** [comment]
Thanks @arpitbbhayani for the assignment. and @gauravrp I am up for some pair programming. HIt me up whenever you want

**Comment by arpitbbhayani** [comment]
Hello @arpitbbhayani,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by ayushsatyam146** [comment]
I am still trying to figure this one out. I will give updates regarding this in the community call 😅

**Comment by arpitbbhayani** [comment]
Hello @arpitbbhayani,

There has been no activity on this issue for the past 5 days.
It would be awesome if you keep posting updates to this issue so that we know you are actively working on it.

We are really eager to close this issue at the earliest, hence if we continue to see the inactivity, we will have to reassign the issue to someone else. We are doing this to ensure that the project maintains its momentum and others are not blocked on this work.

Just drop a comment with the current status of the work or share any issues you are facing. We can always chip in to help you out.

Thanks again.

**Comment by ayushsatyam146** [comment]
I am currently working on a migration issue which I will wrap in a day or 2, after that I will actively work on this. Thanks for the patience.

**Comment by arpitbbhayani** [comment]
https://github.com/DiceDB/dice/pull/1213

**Comment by manishrw** [comment]
Hey @arpitbbhayani , are we doing this anytime soon?
I can pick it up and can reopen the PR #1213
